frontend/src/features/admin-form/create/logic/components/LogicContent/EditCondition/EditConditionBlock.tsx

```diff
@@ -0,0 +1,353 @@
+import { useCallback, useEffect, useMemo } from 'react'
+import {
+  Controller,
+  ControllerRenderProps,
+  UseFormReturn,
+} from 'react-hook-form'
+import { BiTrash } from 'react-icons/bi'
+import {
+  Box,
+  Flex,
+  FormControl,
+  Grid,
+  Stack,
+  VisuallyHidden,
+} from '@chakra-ui/react'
+import { Dictionary, get, pickBy, range } from 'lodash'
+
+import { LOGIC_MAP } from '~shared/modules/logic'
+import { BasicField } from '~shared/types/field'
+import { LogicIfValue, LogicType } from '~shared/types/form'
+
+import { useHasChanged } from '~hooks/useHasChanged'
+import { useWatchDependency } from '~hooks/useWatchDependency'
+import { convertToStringArray } from '~utils/stringFormat'
+import { MultiSelect, SingleSelect } from '~components/Dropdown'
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import IconButton from '~components/IconButton'
+import NumberInput from '~components/NumberInput'
+
+import { BASICFIELD_TO_DRAWER_META } from '~features/admin-form/create/constants'
+import { getIfLogicType } from '~features/logic/utils'
+
+import { EditLogicInputs, FormFieldWithQuestionNumber } from '../../../types'
+
+import { BlockLabelText } from './BlockLabelText'
+
+export interface EditConditionBlockProps {
+  index: number
+  isLoading: boolean
+  handleRemoveLogic?: (index?: number | number[] | undefined) => void
+  formMethods: UseFormReturn<EditLogicInputs>
+  logicableFields: Dictionary<FormFieldWithQuestionNumber> | null
+  mapIdToField: Record<string, FormFieldWithQuestionNumber> | null
+}
+
+export const EditConditionBlock = ({
+  index,
+  isLoading,
+  handleRemoveLogic,
+  formMethods,
+  logicableFields,
+  mapIdToField,
+}: EditConditionBlockProps): JSX.Element => {
+  const name = useMemo(() => `conditions.${index}` as const, [index])
+
+  const {
+    watch,
+    formState: { errors },
+    resetField,
+    register,
+    setValue,
+    control,
+  } = formMethods
+  const ifFieldIdValue = watch(`${name}.field`)
+  const hasFieldIdChanged = useHasChanged(
+    ifFieldIdValue,
+    /* isIgnoreUndefined= */ true,
+  )
+  const conditionStateValue = watch(`${name}.state`)
+  const ifValueTypeValue = watch(`${name}.ifValueType`)
+  const logicTypeValue = watch('logicType')
+  const showValueWatch = useWatchDependency(watch, 'show')
+  const currentSelectedField = useMemo(() => {
+    if (!ifFieldIdValue || !mapIdToField) return
+    return mapIdToField[ifFieldIdValue]
+  }, [ifFieldIdValue, mapIdToField])
+
+  /**
+   * Effect to reset the field if the field to apply a condition on is changed.
+   */
+  useEffect(() => {
+    if (hasFieldIdChanged) {
+      resetField(`${name}.value`, { defaultValue: '' })
+      resetField(`${name}.state`)
+    }
+  }, [hasFieldIdChanged, name, resetField])
+
+  useEffect(() => {
+    if (!currentSelectedField) {
+      resetField(`${name}.ifValueType`)
+      return
+    }
+    setValue(
```

Comment:
Ok. Will derive from current logic block shape instead

Reply:
just a suggestion haha feel free to not do it

Reply:
Looking deeper into this; we don't need to change anything in EditConditionBlock, since the virtual `ifValueType` input is reset/set with useEffect depending on the current field type.

To deprecate the saving of ifValueType in the backend, all we need to do is remove it from the schema eventually